libexec/python/shell.py

```diff
@@ -120,9 +120,9 @@ def parse_image_uri(image, uri=None, quiet=False):
     # Now look for registry, namespace, repo
     image = image.split('/')
 
-    if len(image) == 3:
+    if len(image) > 2:
         registry = image[0]
-        namespace = image[1]
+        namespace = "/".join(image[1:-1])
         repo_name = image[2]
 
```

Comment:
can we be sure that the string at index 2 is reliably the repo name? What if it looks like:

```
In [32]: image
Out[32]: ['registry:4000', 'hubby', 'two', 'imagename']

In [33]: len(image)>2
Out[33]: True

In [34]: image[0] #registry
Out[34]: 'registry:4000'

In [35]: image[2] #repo name?
Out[35]: 'two'

In [36]: image[-1] #repo name
Out[36]: 'imagename'

In [37]: image[1:-1] #namespace
Out[37]: ['hubby', 'two']
```

maybe to be consistent with referencing the namespace from index 1:-1 we should do the same for repo_name and use index -1? If tag/digest is removed, is there any chance the last in the list wouldn't be the reponame?